ex5B/ex5.js

```diff
@@ -10,9 +10,46 @@
  * firstUniqueChar("aaaaaaaaa") === null
  * firstUniqueChar("a") === "a"
  */
-
 function firstUniqueChar(str) {
 
+        var charList = {}, firstUniqueCharVal = null;
+
+        for (var searchIndex = 0; searchIndex < str.length; searchIndex++) {
+
+                if (charList[str.charAt(searchIndex)] === undefined) {
+
+			// Creat new property
+			charList[str.charAt(searchIndex)] = 1;
+
+                } else {
+			// Add to property
+                        charList[str.charAt(searchIndex)] += 1;
+
+                }
+        }
+
+        for (searchIndex = 0; searchIndex < Object.keys(charList).length; searchIndex++) {
+
+                if (charList[Object.keys(charList)[searchIndex]] === 1) {
+
+                        if (firstUniqueCharVal === null) {
+
+                                firstUniqueCharVal = Object.keys(charList)[searchIndex];
+
+                       } else {
+
+                                firstUniqueCharVal = null;
```

Comment:
I think it's probably will be easier to understand if you simply `return null` here instead of resetting the `firstUniqueCharVal` to null, then break, and return it. Then, you can also rename `firstUniqueCharVal` to `firstChar` (which is simpler to understand what it does)